providers/level.js

```diff
@@ -0,0 +1,146 @@
+const { sep, resolve } = require("path");
+const fs = require("fs-nextra");
+const level = require("level");
+const promisify = require("then-levelup");
+
+let baseDir;
+let loaded;
+
+/* eslint-disable no-confusing-arrow */
+exports.init = (client) => {
+  baseDir = resolve(client.clientBaseDir, "bwd", "provider", "level");
+  loaded = new client.method.Collection();
+  fs.readdir(baseDir)
+    .then(files => Promise.all(files.map(async (file) => {
+      const db = await promisify(level(baseDir + sep + file));
+      loaded.set(file, db);
+      return file;
+    })));
+  fs.ensureDir(baseDir).catch(err => client.emit("log", err, "error"));
+};
+
+/* Table methods */
+
+/**
+ * Checks if a directory exists.
+ * @param {string} table The name of the table you want to check.
+ * @returns {Promise<boolean>}
+ */
+exports.hasTable = async table => loaded.has(table);
+
+/**
+ * Creates a new directory.
+ * @param {string} table The name for the new directory.
+ * @returns {Promise<Void>}
+ */
+exports.createTable = async (table) => {
+  const db = await promisify(level(baseDir + sep + table));
+  loaded.set(table, db);
+  return null;
+};
+
+/**
+ * Recursively deletes a directory.
+ * @param {string} table The directory's name to delete.
+ * @returns {Promise<Void>}
+ */
+exports.deleteTable = table => this.hasTable(table)
+  .then(exists => exists ? level.destroy(baseDir + sep + table).then(() => fs.remove(baseDir + sep + table)) : null);
+
+/* Document methods */
+
+/**
+ * Get all documents from a directory.
+ * @param {string} table The name of the directory to fetch from.
+ * @returns {Promise<Object[]>}
+ */
+exports.getAll = async (table) => {
+  if (!loaded.has(table)) return {};
+  return new Promise((res) => {
+    const db = loaded.get(table);
+    const returnValues = {};
+    const stream = db.keyStream();
+    stream.on("data", async (key) => {
+      const value = await db.get(key);
+      try {
+        returnValues[key] = JSON.parse(value);
+      } catch (e) {
+        returnValues[key] = value;
+      }
+    });
+    stream.on("end", () => {
+      res(returnValues);
+    });
+  });
+};
+
+/**
+ * Get a document from a directory.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @returns {Promise<?Object>}
+ */
+exports.get = async (table, document) => loaded.get(table)[document];
+
+/**
+ * Get a random document from a directory.
+ * NOT IMPLEMTABLE IN THIS PROVIDER!
+ * @param {string} table The name of the directory.
+ * @returns {Promise<Object>}
+ */
+exports.getRandom = () => null;
```

Comment:
We're not caching results which means I would need to re-load every single result, grab a random, and let node garbage collect all these results. isn't that super bad in terms of performance? why do we need random anyway?